stan/math/prim/mat/fun/gp_exp_quad_cov.hpp

```diff
@@ -275,11 +275,11 @@ gp_exp_quad_cov(const std::vector<T_x1> &x1, const std::vector<T_x2> &x2,
     return cov;
   }
 
-  for (size_t i = 0; i < x1_size; ++i) {
-    check_not_nan(function_name, "x1", x1[i]);
+  for (auto &&x1_i : x1) {
```

Comment:
> These should be using a vectorized check_not_nan so that the index can also be printed and we don't have all this boilerplate looping.

Agree this should use a vectorized check_nan, but the vectorized version of `check_not_nan` does not work for vectors of eigen matrices atm :-(

After Andrew and I sort out the more generic templating discussion in #1425 then I'm going to come back to these check functions and clean them up so we can do that.